backend/src/apiserver/server/experiment_server.go

```diff
@@ -10,14 +10,66 @@ import (
 	"github.com/kubeflow/pipelines/backend/src/apiserver/resource"
 	"github.com/kubeflow/pipelines/backend/src/common/util"
 	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
 )
 
+// Metric variables. Please prefix the metric names with experiment_server_.
+var (
+	// Used to calculate the request rate.
+	createExperimentRequests = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "experiment_server_create_requests",
+		Help: "The total number of CreateExperiment requests",
+	})
+
+	getExperimentRequests = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "experiment_server_get_requests",
+		Help: "The total number of GetExperiment requests",
+	})
+
+	listExperimentRequests = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "experiment_server_list_requests",
+		Help: "The total number of ListExperiments requests",
+	})
+
+	deleteExperimentRequests = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "experiment_server_delete_requests",
+		Help: "The total number of DeleteExperiment requests",
+	})
+
+	archiveExperimentRequests = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "experiment_server_archive_requests",
+		Help: "The total number of ArchiveExperiment requests",
+	})
+
+	unarchiveExperimentRequests = promauto.NewCounter(prometheus.CounterOpts{
+		Name: "experiment_server_unarchive_requests",
+		Help: "The total number of UnarchiveExperiment requests",
+	})
+
+	// TODO(jingzhang36): error count and success count.
+
+	experimentCount = promauto.NewGauge(prometheus.GaugeOpts{
+		Name: "experiment_server_run_count",
+		Help: "The current number of experiments in Kubeflow Pipelines instance",
+	})
```

Comment:
How does this get initial experiment count? I'm only seeing updates

Reply:
Initialization is done when the metric variables are declared https://www.robustperception.io/why-predeclare-metrics